lab3/src/MyScanner.java

```diff
@@ -0,0 +1,213 @@
+import java.io.*;
+import java.security.NoSuchAlgorithmException;
+import java.util.*;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class MyScanner {
+    File file = new File("token.in.txt");
+    File p1 = new File("p1-flcd.txt");
+    public List<String> operators = new ArrayList();
+    public List<String> separators = new ArrayList();
+    public List<String> reservedWords = new ArrayList();
+    public PIF pif = new PIF();
+    public SymbolTable symbolTableIdentifiers = new SymbolTable();
+    public SymbolTable symbolTableConstants = new SymbolTable();
+
+
+    public void readTokenFile(String filename) {
+        //read the token.in file and add each operator and separator to it s list
+        List<String> lines = new ArrayList<>();
+        BufferedReader reader;
+        try {
+            reader = new BufferedReader(new FileReader(filename));
+            String line = reader.readLine();
+            while (line != null) {
+                lines.add(line);
+                line = reader.readLine();
+            }
+            reader.close();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        separators.add("\t");
+        for (int i = 0; i <= lines.size(); i++) {
+            if (i <= 10) {
+                operators.add(lines.get(i));
+            }
+            if (i > 10 && i <= 18) {
+                separators.add(lines.get(i));
+            }
+            if (i > 18 && i <= 32) {
+                reservedWords.add(lines.get(i));
+            }
+
+        }
+    }
+
+    private List<String> readFile(String filename) {
+        //read every file and return a list with the lines
+        List<String> lines = new ArrayList<>();
+        BufferedReader reader;
+        try {
+            reader = new BufferedReader(new FileReader(filename));
+            String line = reader.readLine();
+            while (line != null) {
+                lines.add(line);
+                line = reader.readLine();
+            }
+            reader.close();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+
+        return lines;
+    }
+
+    public List<String> tokenize(String line, int lineCount) {
+        List<String> tokens = new ArrayList<>();
```

Comment:
here the separators and the operators could've been used instead of the hard coded string " ,()[]{};+-*/%=><!\t"